internal/sftp/client.go

```diff
@@ -14,4 +14,7 @@ type Client interface {
 	// Upload transfers data from the provided source reader to a specified
 	// destination on the SFTP server.
 	Upload(ctx context.Context, src io.Reader, dest string) (bytes int64, err error)
+	// Delete removes data from the provided dest to a specified
+	// destination on the SFTP server.
+	Delete(ctx context.Context, dest string) (bytes int64, err error)
```

Comment:
I can't think of any reason we'd need to know the size of the file after deleting it - I think the `bytes` return value should be removed.